measurements.py

```diff
@@ -46,6 +47,112 @@ def compute_low(data: AnalogSpan, mid: float):
     return mode(x for x in data if x < mid)
 
 
+def find_three_closest(nums: List[Tuple[int, bool]], target: int):
+    # If the array has 3 or fewer elements, return the entire array
+    if len(nums) <= 3:
+        return nums
+
+    # Calculate the difference from the target for each number
+    diff_list = [abs(num[0] - target) for num in nums]
+
+    # Get the indices of the three smallest differences
+    # This maintains the original order of the elements
+    idx_three_closest = sorted(
+        range(len(diff_list)), key=lambda i: diff_list[i])[:3]
+
+    # Select the numbers at these indices, maintaining original order
+    closest_three = [nums[i] for i in sorted(idx_three_closest)]
+
+    return closest_three
+
+
+def find_starting_location(data, mid, hysteresis) -> Optional[int]:
+    """Find a starting location that is outside of the hysteresis band."""
+
+    start_index = int(len(data) / 2)
+    start_value = data[start_index]
+    if (abs(start_value - mid) < hysteresis):
+        right_lt_index = data.find_lt(mid - hysteresis, start=start_index)
+        right_gt_index = data.find_gt(mid + hysteresis, start=start_index)
+        left_lt_index = data.rfind_lt(mid - hysteresis, end=start_index)
+        left_gt_index = data.rfind_gt(mid + hysteresis, end=start_index)
+        # find the nearest sample index that is outside of the hysteresis range.
+        indexes = [right_lt_index, right_gt_index,
+                   left_lt_index, left_gt_index]
+        valid_indexes = [v for v in indexes if v is not None]
+        if not valid_indexes:
+            # in this case, no sample in the data is outside of the hysteresis zone.
+            # since hysteresis is always smaller than the amplitude, this should only happen for a constant value signal.
+            return None
+        closest_index = min(
+            valid_indexes, key=lambda x: abs(x - start_index))
```

Comment:
You can also write this as:
`        closest_index = min(abs(x - start_index) for x in valid_indexes)`